csrc/layernorm_quant_kernels.cu

```diff
@@ -227,8 +241,22 @@ void fused_add_rms_norm_static_fp8_quant(
   bool ptrs_are_aligned =
       inp_ptr % 16 == 0 && res_ptr % 16 == 0 && wt_ptr % 16 == 0;
   if (ptrs_are_aligned && hidden_size % 8 == 0) {
-    LAUNCH_FUSED_ADD_RMS_NORM(8);
+    if (is_fp8_ocp()) {
+      LAUNCH_FUSED_ADD_RMS_NORM(8, c10::Float8_e4m3fn);
+    }
+#ifdef USE_ROCM
+    else {
+      LAUNCH_FUSED_ADD_RMS_NORM(8, c10::Float8_e4m3fnuz);
+    }
+#endif
   } else {
-    LAUNCH_FUSED_ADD_RMS_NORM(0);
+    if (is_fp8_ocp()) {
+      LAUNCH_FUSED_ADD_RMS_NORM(0, c10::Float8_e4m3fn);
+    }
+#ifdef USE_ROCM
+    else {
+      LAUNCH_FUSED_ADD_RMS_NORM(0, c10::Float8_e4m3fnuz);
+    }
+#endif
```

Comment:
Could clean up dispatching logic here

Reply:
The nested dispatch really helped clean this up.